virttest/utils_test/__init__.py

```diff
@@ -1755,46 +1755,49 @@ def __str__(self):
 
-        def _parameters_filter(stress_type):
-            """Set parameters according stress_type"""
-            _control_files = {'unixbench': "unixbench5.control",
-                              'stress': "stress.control",
-                              'iozone': "iozone.control"}
-            _check_cmds = {'unixbench': "pidof -s ./Run",
-                           'stress': "pidof -s stress",
-                           'iozone': "pidof -s iozone"}
-            _stop_cmds = {'unixbench': "killall ./Run",
-                          'stress': "killall stress",
-                          'iozone': "killall iozone"}
-            try:
-                control_file = _control_files[stress_type]
-                self.control_path = os.path.join(data_dir.get_root_dir(),
-                                                 "shared/control",
-                                                 control_file)
-                self.check_cmd = _check_cmds[stress_type]
-                self.stop_cmd = _stop_cmds[stress_type]
-            except KeyError:
-                self.control_path = ""
-                self.check_cmd = ""
-                self.stop_cmd = ""
+        :param vm: the vm to be loading stress
+        :param stress_type: the name of the stress tool
+        :param download_url: from where download the stress tool
+        :param make_cmds: make command of the stress tool
+        :param stress_cmds: the command to launch stress,
+        use stress_type instead if not defined
+        :param stress_args: the arguments of the stress tool
+        :param work_path: the relative work path of the stress tool,
+        e.g. for iozone: "src/current"
+        :param check_installed_cmds: the command to check if the tool is installed,
+        skip the check if not defined.
+        :param download_type: currently support "git" or "tarball"
+        """
 
         self.vm = vm
         self.params = vm.params
         self.timeout = 60
         self.stress_type = stress_type
-        if stress_type not in ["stress", "unixbench", "iozone"]:
-            raise StressError("Stress %s is not supported now." % stress_type)
-
-        _parameters_filter(stress_type)
-        self.stress_args = self.params.get("stress_args", "")
+        stress_cmds = stress_cmds or stress_type
+        self.stress_cmds = self.params.get('stress_cmds_%s' % stress_type, stress_cmds)
+        self.stress_args = self.params.get("%s_args" % stress_type, stress_args)
+        self.download_url = self.params.get('download_url_%s' % stress_type,
+                                            download_url)
+        self.download_type = download_type
+        self.base_name = self.download_url.split("/")[-1]
+        self.make_cmds = self.params.get('make_cmds_%s' % stress_type, make_cmds)
+        self.work_path = self.params.get('%s_work_path' % stress_type, work_path)
+        self.check_cmd = "pidof -s %s" % stress_type
+        self.stop_cmd = "killall %s" % stress_type
```

Comment:
can we use `"pkill -9 %s" % stress_type` to force the stress process to kill ? what would you say